### DISCUSSION TITLE
Data Importer Configuration Problem

### DISCUSSION THREAD

**Discussion by z0rg0n** [firefly-iii/firefly-iii#9184] [Firefly III]
I got both fireflyiii and the data importer up but when I add my OAuth client ID to the Data Importer I get the error:
`ERR_CONNECTION_REFUSED`

I have done a search for similar discussions and did find some relevant threads.
1. https://github.com/orgs/firefly-iii/discussions/9037
This person seems to have found a solution but doesn't document it well. They're also on a different platform.

2.  https://github.com/orgs/firefly-iii/discussions/8866
A solution is suggested by JC5 and a solution is found by the author. However, changing my .importer.env file with `VANITY_URL=http://data-importer.firefly3.server.local` and `http://firefly3.omv.local` as described  give the errors:
`ERR_NAME_NOT_RESOLVED`

This is probably a configuration error since this is my first time using Docker. Below are my configuration files.

[docker-compose.yml](https://pastebin.com/7jT3EfmX)
[.env](https://pastebin.com/c9xP4g6P)
[importer.env](https://pastebin.com/nSBW9vAj)
[.db.env](https://pastebin.com/LJARDzXJ)

I'm not sure where to go from here. I've exhausted all resources I can find.

**Comment by JC5** [comment]
I think the `FIREFLY_III_URL` has to be "http://app:8080", try that :)

**Comment by z0rg0n** [comment]
That's what it was in my first few attempts.

I changed `FIREFLY_III_URL=http://app:8080` in the .importer.env file in a new VM and got the same result, `ERR_CONNECTION_REFUSED`.

**Reply by JC5** [reply]
Alright, so could you share a log or some more details beyond "it says connection refused?"

**Comment by z0rg0n** [comment]
I apologize. I didn't see log files in the GUI so I had to do some learning on docker.

Here are the four configuration files:
[.importer.env](https://pastebin.com/5UCufeh6)
[.env](https://pastebin.com/RAkVC29M)
[.db.env](https://pastebin.com/ej27gH2y)
[docker-compose.yml](https://pastebin.com/6QLb4ttg)

I run this command in the directory with these files:
`sudo docker compose -f docker-compose.yml up -d --pull=always`

I can then see both the fireflyiii core and dataimporter at http://192.168.1.248:666/ and http://192.168.1.248:667/ respectively.

I make an account and at that point [this is the log of the dataimporter](https://pastebin.com/TuQERixH).

Next I create a new O-auth token with http://192.168.1.248:667/callback as a redirect URL and confidential unchecked.

I add the number shown in that O-auth client ID to the dataimporter and push submit. The `ERR_CONNECTION_REFUSED` error shows in my browser.

The log file of the dataimporter[ now shows this.](https://pastebin.com/PnCX6sR3) Both the logs files are exactly the same so maybe I'm not understanding something in Docker.

If this helps, the [fireflyiii logs are here.](https://pastebin.com/TZgLM3DS)



Edit:
Just doing some trial and error. Any number in the 1-65536 range that I put into the dataimporter gives the same result `ERR_CONNECTION_REFUSED`. It's maybe like the dataimporter doesn't see fireflyiii.

Edit2:
If it makes a difference I'm running fireflyiii in a Proxmox VM and on a Unify router.

Edit3:
Tried to change the vanity url in the `.importer.env` file to `VANITY_URL=http://localhost:666`. This didn't work either.

**Comment by z0rg0n** [comment]
I tried the same thing in a Ubuntu VM instead of Debian 12 and got the same result.

The easiest thing for me to test next is this router. I'm going to build a new one with pf-sense and see if that fixes the problem.

**Comment by z0rg0n** [comment]
Update:
I set up a brand new router with a different OS and connected my machine with the VM running Fireflyiii to it.

I got the same exact error as before so that rules out the router.

The next thing I'll have to try is picking up another used PC to see if I get the same error on that. It's possibly a waste of money but I haven't got my budgeting set up in Fireflyiii yet so it can't hurt me right?

**Comment by z0rg0n** [comment]
No idea why but changing the ports to the default 80 and 81 fixed the problem.
